src/metabase/lib/temporal_bucket.cljc

```diff
@@ -228,12 +229,11 @@
   #{})
 
 (defn- mark-unit [options option-key unit]
-  (cond->> options
-    (some #(= (:unit %) unit) options)
-    (mapv (fn [option]
-            (cond-> option
-              (contains? option option-key) (dissoc option option-key)
-              (= (:unit option) unit)       (assoc option-key true))))))
+  (mapv (fn [option]
```

Comment:
This change is safe if you check the call site of this function and the function that is calling this function,